pandas/tests/indexes/test_multi.py

```diff
@@ -165,6 +165,18 @@ def test_set_name_methods(self):
         assert res is None
         assert ind.names == new_names2
 
+    def test_multiindex_set_names(self):
+        # GH 21149
+        '''Ensure that .set_names for MultiIndex with
+            nlevels == 1 does not raise any errors
+        '''
+        result = pd.MultiIndex.from_product([[0, 1]])
+        result.set_names('first', level=0, inplace=True)
```

Comment:
test both inplace=True and False with a parameterized test

Reply:
done